x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/model_select.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import { EuiSelectable } from '@elastic/eui';
+
+import { MlModel } from '../../../../../../../common/types/ml';
+import { IndexNameLogic } from '../../index_name_logic';
+import { IndexViewLogic } from '../../index_view_logic';
+
+import { MLInferenceLogic } from './ml_inference_logic';
+import { ModelSelectLogic } from './model_select_logic';
+import { ModelSelectOption, ModelSelectOptionProps } from './model_select_option';
+import { normalizeModelName } from './utils';
+
+export const ModelSelect: React.FC = () => {
+  const { indexName } = useValues(IndexNameLogic);
+  const { ingestionMethod } = useValues(IndexViewLogic);
+  const {
+    addInferencePipelineModal: { configuration },
+  } = useValues(MLInferenceLogic);
+  const { selectableModels, isLoading } = useValues(ModelSelectLogic);
+  const { setInferencePipelineConfiguration } = useActions(MLInferenceLogic);
+
+  const { modelID, pipelineName, isPipelineNameUserSupplied } = configuration;
+
+  const getModelSelectOptionProps = (models: MlModel[]): ModelSelectOptionProps[] =>
+    (models ?? []).map((model) => ({
+      ...model,
+      label: model.modelId,
+      checked: model.modelId === modelID ? 'on' : undefined,
+    }));
+
+  const onChange = (options: ModelSelectOptionProps[]) => {
+    const selectedOption = options.find((option) => option.checked === 'on');
+    setInferencePipelineConfiguration({
+      ...configuration,
+      inferenceConfig: undefined,
+      modelID: selectedOption?.modelId ?? '',
+      fieldMappings: undefined,
+      pipelineName: isPipelineNameUserSupplied
+        ? pipelineName
+        : indexName + '-' + normalizeModelName(selectedOption?.modelId ?? ''),
+    });
+  };
+
+  const renderIndexOption = (option: ModelSelectOptionProps) => <ModelSelectOption {...option} />;
+
+  return (
+    <EuiSelectable
+      data-telemetry-id={`entSearchContent-${ingestionMethod}-pipelines-configureInferencePipeline-selectTrainedModel`}
+      options={getModelSelectOptionProps(selectableModels)}
+      singleSelection="always"
+      listProps={{
+        bordered: true,
+        rowHeight: 90,
```

Comment:
Have you tested if this row height needs to be increased when rendering on mobile?

Reply:
Update: Tested on my end. Looks like this on mobile:

<img width="389" alt="Screenshot 2023-12-01 at 12 08 18 PM" src="https://github.com/elastic/kibana/assets/26927948/5a2ef0c2-cb91-495d-a2a2-09f2c0905126">

Reply:
Yes, this is not pretty. I'm working on making the height depend on the screen size, using [your code](https://github.com/elastic/kibana/pull/172330/files#diff-5e41f63398077a4197271090f06b65da25f912a99606e13cea10f99d1c4f8935R66) as reference. At a minimum I want to make sure all parts are visible and navigable. This is what it looks like - I'll add a tech debt item to fix this properly post-FF, is that alright?

<img width="449" alt="Screenshot 2023-12-01 at 12 13 34" src="https://github.com/elastic/kibana/assets/14224983/aba22548-6773-4f85-9dc6-c4f6b4ecf019">

Reply:
Sounds good, let's fix post-FF

Reply:
I added an item to the tech debt ticket, linking back to this thread.